CommonTools/Utils/src/MethodArgumentSetter.h

```diff
@@ -22,6 +22,11 @@ namespace reco {
       }
       void operator()(const char *begin, const char *end) const {
         assert(begin+1 <= end-1); // the quotes are included in [begin,end[ range.        
+        //in boost 1.67, the parser appends any extra white space to the
+        // end
+        if(*(end-1) != *begin) {
+          --end;
+        }
```

Comment:
do I understand correctly that this can not be ported to another release using another boost version .. and may resurface as a problem if a future version of boost changes behavior?
Perhaps a more explicit check on the boost version is more appropriate.

Reply:
This change will work properly with boost new and old. The updated unit tests to cover the newly discovered problem checks that.

This change makes sure that the end dilimiter matches the beginning delimiter found by the parser. For the old version of boost, or when there is no trailing white space, the if is satisfied immediately.

Reply:
ah. I should assume here already that ```*begin``` is the opening delimiter.
OK.